### DISCUSSION TITLE
age implementation in Dart

### DISCUSSION THREAD

**Discussion by v3rm0n** [FiloSottile/age#419] [General]
We want to use age in a Flutter app (with Yubikey and x25519) so we started by implementing a Dart package called [dage](https://pub.dev/packages/dage).
It's not ready and not fully spec compliant, but it is functional.
We are planning to add the Yubikey x25519 functionality in a separate package as a plugin to this system.
Will there be a generic spec for age plugins at one point? Would be great if we could make our plugin work with all of the implementations.

**Comment by str4d** [comment]
The age plugin spec is here (I need to respond to a few comments, and then we can probably get the draft merged): https://github.com/C2SP/C2SP/pull/5

**Reply by quite** [reply]
Is there any plugin implementation in go yet?

**Comment by v3rm0n** [comment]
Our Dart implementation is now spec compliant and can be used as a Dart library or as an executable. It supports plugins as well but not in the plugin spec compliant way. Probably at one point we will implement that as well. Repo is here if somebody is interested https://github.com/Producement/dage

There's also a plugin for this library that enables the usage of Curve25519 and RSA operations on a Yubikey https://github.com/Producement/age-yubikey-pgp
